### DISCUSSION TITLE
What's the simplest way to dip our toes?

### DISCUSSION THREAD

**Discussion by AngelMunoz** [ScottArbeit/Grace#16] [General]
I'd say as a simple user-like person trying this out, what do I need to get something up and running?

My curiosity driven use case is to setup/mount this in a Raspberry PI server at home just to start toying with it and eventually spin up a VM in the cloud just to have a better grasp of it.

Given this README fragment:

> - Grace.CLI is the command-line interface for Grace.
> - Grace.Server is an ASP.NET Core project that defines the Web API for Grace.
> - Grace.SDK is a .NET class library that is a platform-specific projection of the Web API; ...

- As user-like person:
  - _What_ do I need to install _where_?
  - Which project goes on the "server" and which goes on my local machine?

Seems fairly straight forward but I don't want to take any assumptions to avoid confusion.

- As a curious dev, if I want to play a little with a custom client for this, which project serves me better?

I'd say the Server is likely the one I need to figure out setup to be able to write a toy client against it. That being said:

- which is preferred for the toy client the server or the sdk?

-----

Also great NDC talk, project and motivations, this looks quite amazing and I'm hoping for the best, Kudos to you!

**Comment by ScottArbeit** [comment]
@AngelMunoz Thank you so much for your interest! and I really appreciate the detailed questions.

And, yes, I'm behind on this for sure. So far, I've been the only one working on it, so it's unfortunately customized to my machine. I've started working on making it nicer for everyone else, and I hope to have some way for you to get started in the next week.

I will say: one thing that's true about Grace that I'm going to fix soon is: right now it requires CosmosDB as the database. When I had to write that part, it was before Dapr supported a query API that worked across multiple state stores. Now they do, and I need to update ~10 functions or so to use the new Dapr Query API, and then we'll be able to test with containerized or cloud-based CockroachDB, MongoDB, and Redis, as well as CosmosDB. They are [the current Dapr state stores that support both Actors and Queries](https://docs.dapr.io/reference/components-reference/supported-state-stores/), but that list will continue to grow over time, and Grace will be able to take advantage of it.

I'll let you know when there's some movement on it, and I'll answer your questions when I do.

**Reply by AngelMunoz** [reply]
Sounds great!
Yeah no worries, I get it these things take time I'll keep an eye here for sure!

**Comment by mrc-eu** [comment]
Watched your presentation on NDC yesterday and got me really interested in Grace and F#. The fact that you said that you hated Electron helped a lot in making me star your project ;).

What is your realistic timeline for getting Grace production ready? 1, 2, 3 years?
